lib/evss/disability_compensation_form/service.rb

```diff
@@ -50,7 +78,18 @@ def submit_form526(form_content)
         with_monitoring_and_error_handling do
           headers = { 'Content-Type' => 'application/json' }
           options = { timeout: Settings.evss.disability_compensation_form.submit_timeout || 355 }
-          raw_response = perform(:post, 'submit', form_content, headers, options)
+          begin
+            raw_response = perform(:post, 'submit', form_content, headers, options)
+          rescue Net::ReadTimeout, Timeout::Error, Faraday::TimeoutError => e
```

Comment:
Hmmm, theoretically I don't think this `rescue` should ever fire for `Timeout::Error` and `Faraday::TimeoutError`.
The `rescue` in `lib/common/client/base` should catch them and raise `Common::Exceptions::GatewayTimeout`.
So the timeout errors never make it to this level. There's [a spec for this scenario](https://github.com/department-of-veterans-affairs/vets-api/blob/d28cc999502a96a58f60378e415deb64a4290448/spec/lib/evss/disability_compensation_form/service_spec.rb#L30).

However, as I pointed out, there are a bunch of timeout errors in Sentry.  Will this catch them?  Or will they not get rescued for the same reason their getting past `common/client/base`?  I don't know.
Maybe play around with some code like [this SO answer](https://stackoverflow.com/questions/56649234/how-i-can-generate-faraday-timeout-error-manually-in-ruby-on-rails).

Also, you're trying to log `raw_response` in this rescue. I believe it's value will be `nil`.  It never gets set since an error occurred inside the method that is supposed to return it's value.  Thinking about it broadly, if you get a timeout error, there is no response to analyze.  None was received because the request timed out.

Any rescuing/logging you do in this file, you have to account for any manipulation (`rescue`/`raise` stuff) that happens in `common/client/base`.  Likewise anything that happens in `common/client/base` is dependent on what happens in the Faraday middlewares. That's the closest you can get to the actual request/response.  That's probably why the `:logging` middleware was written.

Reply:
>Hmmm, theoretically I don't think this rescue should ever fire for Timeout::Error and Faraday::TimeoutError.
The rescue in lib/common/client/base should catch them and raise Common::Exceptions::GatewayTimeout.

ugh. I forgot that all the network calls were going through there.

>Also, you're trying to log raw_response in this rescue. I believe it's value will be nil. It never gets set since an error occurred inside the method that is supposed to return it's value. Thinking about it broadly, if you get a timeout error, there is no response to analyze. None was received because the request timed out.

this was just for completeness sake, but also, i don't know if something beyond EVSS could timeout, and then EVSS would really return a response (that describes a further upstream timeout). don't know. just wanted to cover my bases